src/foam/dao/index/Index.js

```diff
@@ -15,52 +15,114 @@
  * limitations under the License.
  */
 
-/* Indexed Memory-based DAO. */
+/**
+  The Index interface for an ordering, fast lookup, single value,
+  index multiplexer, or any other MDAO select() assistance class.
 
-/*
- * TODO:
- *  update(oldValue, newValue)
- *  reuse plans
- *  add ability for indices to pre-populate data
- */
+  Each Index subclass also defines an IndexNode class. Index defines
```

Comment:
Is this over specified? Do we need to know that Indices have nodes? Isn't this an implementation detail?